### PR TITLE
Document GitRepository source in HelmRelease guide

### PR DESCRIPTION
**Pull request by hiddeco**

### CODE REVIEW

File: docs/guides/helmreleases.md [hunk 1]
```diff
@@ -48,11 +59,73 @@ The `url` can be any HTTP/S Helm repository URL.
     HTTP/S basic and TLS authentication can be configured for private
     Helm repositories. See the [`HelmRepository` CRD docs](../components/source/helmrepositories.md)
```
Comment (dholbach) [inline docs/guides/helmreleases.md:60]: Does this link work for you?

File: docs/guides/helmreleases.md [hunk 2]
```diff
@@ -48,11 +59,73 @@ The `url` can be any HTTP/S Helm repository URL.
...
+
+The `ignore` defines file and folder exclusion for the
+artifact produced, and follows the [`.gitignore` pattern
+format](https://git-scm.com/docs/gitignore#_pattern_format).
+The above example only includes the `charts` directory of the
+repository and omits all other files.
+
+!!! hint "Authentication"
+    HTTP/S basic and SSH authentication can be configured for private
+    Git repositories. See the [`GitRepository` CRD docs](../components/source/gitrepositories.md)
```
Comment (dholbach) [inline docs/guides/helmreleases.md:122]: Does this link work for you?

**Review by dholbach — APPROVED** [review]
The text generally reads fine, I wasn't able to test the instructions though.

General :+1: - maybe check the links to the CRD docs again?

File: docs/guides/helmreleases.md [hunk 3]
```diff
@@ -48,11 +59,73 @@ The `url` can be any HTTP/S Helm repository URL.
...
+kind: GitRepository
+metadata:
+  name: podinfo
+  namespace: gitops-system
+spec:
+  interval: 1m
+  url: https://github.com/stefanprodan/podinfo
+  ref:
+    branch: master
+  ignore: |
```
Comment (scottrigby) [inline docs/guides/helmreleases.md:95]: feat: add `--ignore` flag to `gotk create source git`? (I see a few other struct fields missing too, just noting this as I'm seeing it in the doc).

Speaking of, do we want corresponding `gotk` commands linked from these yaml definitions in general?

Comment (hiddeco) [reply docs/guides/helmreleases.md:95]: > feat: add `--ignore` flag to `gotk create source git`? (I see a few other struct fields missing too, just noting this as I'm seeing it in the doc).

Yep, focus has been to get the API stable first before adding every possible option to the CLI.

> Speaking of, do we want corresponding `gotk` commands linked from these yaml definitions in general?

In the future, probably, the [installation guide](https://toolkit.fluxcd.io/guides/installation/) already does this.

**Pull request merged**